10-JavaScript/variables-conditionals-iteration.md

```diff
@@ -0,0 +1,308 @@
+
+`var` and `let` declare variables that can be re-assigned. `var` and `let` mean subtly different things, both having to do with [scope](https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var-to-declare-a-variable) which we'll get to later.  **A good rule of thumb and our best practice moving forward is to always use `let` for variables we expect to be re-assigned.**
+
+It may make sense to use `var` if you are working with an older version of the JavaScript interpreter.
+
+### We Prefer `const` for All Other Variables
+
+Constants cannot change their value through reassignment. **Our best practice moving forward is to always use `const` for variables we expect to not be re-assigned.**
+
+In your Node REPL, input these lines. Do you get what you expect?
+
+1. `var dees_favorite_color = 'green';`
+1. `dees_favorite_color;`
+1. `let dees_favorite_number = 72;`
+1. `dees_favorite_number;`
+1. `const dees_favorite_game = 'night in the woods';`
+1. `dees_favorite_game;`
+1. `dees_favorite_number = 16;`
+1. `let dees_favorite_number;`
+1. `dees_favorite_game = 'butterfly soup';`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `undefined`
+  1. `'green'`
+  1. `undefined`
+  1. `72`
+  1. `undefined`
+  1. `'night in the woods'`
+  1. `16`
+  1. `Thrown: SyntaxError: Identifier 'dees_favorite_number' has already been  declared` You only need to do the variable declaration (using `let` or `const`) once. You can just re-assign a `let` variable without the keyword `let`.
+  1. `Thrown: TypeError: Assignment to constant variable.` You can't re-assign a `const` variable.
+
+</details>
+
+### Always Declare!
+
+Declaring a variable without one of these three keywords creates a global variable, which is bad practice and leads to unintended consequences-- don't do it!
+
+## Playing with Multi-Line JavaScript
+
+We are now in the zone where what we want to learn about JavaScript is going to take multiple lines, so working with the Node REPL will be frustrating.
+
+From now on:
+
+1. We'll be making JavaScript files with a `.js` extension
+1. Running these files from the terminal with `$ node file.js`
+1. Printing to the terminal using `console.log()` (much like `puts`)
+
+Observe the following:
+
+```bash
+$ touch intro-to-javascript.js
+```
+
+Copy and paste the following code into `intro-to-javascript.js`:
+
+```javascript
+const testVariable = 'Hello World!';
+console.log(testVariable);
+```
+
+```bash
+$ node intro-to-javascript.js
+```
+
+## Conditionals
+
+[We can do `if ... else` logic in JavaScript!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else)
+
+We use the keywords `if`, `else if`, and `else`. Note `else if` instead of `elsif` like in Ruby!
+
+Conditional expressions are surrounded by parenthesis `()` and each block is surrounded by brackets `{}`.
+
+To test equality between strings, we use `===`. Note triple instead of `==` like in Ruby!
```

Comment:
Do we want to mention that `==` works in JS but will give slightly wrong answers?  I feel like if I was reading this I'd expect `==` to be a syntax error.

Reply:
Hmm. I'll think on this. Historically, we've just... never spent time talking about the difference in JS? So it feels weird/bad to me to make more time for it, when historically we haven't, but yeah I super love the idea of bringing it up.
Other considerations: is it important to bring up on day one? I could argue either way

Reply:
I don't know we need to get into it on day one I just think mentioning that `==` exists but does something else could make sense but I don't feel strongly one way or the other.